src/goto-symex/symex_assign.cpp

```diff
@@ -67,6 +67,31 @@ expr_skeletont expr_skeletont::compose(expr_skeletont other) const
   return expr_skeletont(apply(other.skeleton));
 }
 
+optionalt<expr_skeletont> expr_skeletont::array() const
+{
+  if(skeleton.id() != ID_index)
+    return {};
+  return expr_skeletont{to_index_expr(skeleton).array()};
+}
+
+optionalt<expr_skeletont> expr_skeletont::compound() const
```

Comment:
Name this `member_compound`, as I would expect a method named `compound` to also return the array of an `index_exprt`